src/test/java/com/connexta/search/query/QueryEndpointTests.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.search.query;
+
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.connexta.search.query.controllers.QueryController;
+import com.connexta.search.query.exceptions.IllegalQueryException;
+import com.connexta.search.query.exceptions.MalformedQueryException;
+import com.connexta.search.query.exceptions.QueryException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.stream.Stream;
+import javax.inject.Inject;
+import org.apache.http.client.utils.URIBuilder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+@WebMvcTest(QueryController.class)
+public class QueryEndpointTests {
+
+  public static final String QUERY_STRING = "id=12efab35fab21afdd8932afa38951aef";
+  public static final String URI_QUERY_PARAMETER = "q";
+  public static final String SEARCH_ENDPOINT = "/search";
+
+  @MockBean private QueryManager queryManager;
+
+  @Inject private QueryController queryController;
+
+  @Inject private MockMvc mockMvc;
+
+  private static Stream<Arguments> requestsThatThrowErrors() {
+    return Stream.of(
+        Arguments.of(500, new QueryException(HttpStatus.INTERNAL_SERVER_ERROR, "Test")),
+        Arguments.of(400, new IllegalQueryException(Arrays.asList("Test"))),
+        Arguments.of(400, new MalformedQueryException(new RuntimeException())));
+  }
+
+  @Test
+  public void testQueryManagerReturnsList() throws Exception {
+    given(queryManager.find(QUERY_STRING)).willReturn(new ArrayList<>());
```

Comment:
@jhunzik Why do use a new Mockito here? Is this way better? Would it be easy to switch all of the tests to use this?

Reply:
I'm not sure what you're referring to. This was just just to show the "happy path" when the QueryManager works and when the service should return 200 OK. I figured returning a real empty List would work well for that. If that's what you're talking about, I think I remember having issues with returning a mocked out ArrayList.

Reply:
Sorry for the confusion. Why do you use `org.mockito.BDDMockito` instead of `org.mockito.Mockito`?

|                          |                                                                                                                     |
|--------------------------|---------------------------------------------------------------------------------------------------------------------|
| `org.mockito.BDDMockito` | ```final List queryResults = new ArrayList<>(); given(queryManager.find(QUERY_STRING)).willReturn(queryResults);``` |
| `org.mockito.Mockito`    | ```final List queryResults = new ArrayList<>(); when(queryManager.find(QUERY_STRING)).thenReturn(queryResults);```  |

We use `org.mockito.Mockito` everywhere else in our tests.

Reply:
Ah that was a mistake. The new Mockito just provides a way to write tests in a behavior driven way, similar to  how we would write it in Spock.